### DISCUSSION TITLE
Limit registrations?

### DISCUSSION THREAD

**Discussion by dedene** [zitadel/zitadel#5691] [Q&A]
Is it possible to use Zitadel as SSO for internal use but allow self registration for employees only? I.e. use Github IdP but limit it to a certain organisation? Or use Google IdP but certain domain only?

**Comment by hifabienne** [comment]
Hei @dedene 
Most of our policies have a default configuration on the instance level, and can be overwritten on a certain organization.
You can disable the "allow registration" setting in the login policy of the organization where you don't want to allow self registration.

**Reply by dedene** [reply]
Hi @hifabienne,

Thx for your reply! I've seen you can disable registration. But what I'm actually looking to accomplish is to use Zitadel as Identity Broker for multiple sources (mainly Google Workspace and Github) but only allow users from our Github Organization or Google Workspace account to register. Is this possible i.e. using policies?

**Reply by hifabienne** [reply]
At the moment it is not possible on the side of ZITADEL to make any restrictions per configuration.
I have not tried.  But I think you could use the ZITADEL ACtions.
ZITADEL actions are custom code written in javascript you can run at certain workflow points. in your case this would be external authentication - pre creation. you could add a skript that checks if the user is from your organization.

**Reply by dedene** [reply]
I was indeed also thinking/hoping something like this would be possible using actions. However on https://zitadel.com/docs/apis/actions/external-authentication i.e. I do not see a method or API for stopping a flow or preventing a user to register (i.e. in the pre creation flow). Do you have an example of that? Or any idea how that would work?

**Reply by swratten** [reply]
The company I work for is also interested in this, as it will help scope down what is possible for registration purposes, as currently it appears registration is either open to all or restricted to all.

The ideal scenario for us is as follows.

Have "Registration allowed" but scoped down to optionally:

- Self-Register with email, but **limited** to Configured Domains / configurable per domain

- Admins / users with sufficient permissions may create users with any valid email

- External IDP provided users

- And of course option to have an open registration if needed

**Comment by hifabienne** [comment]
Hei, you do have now the possibility to enable the registration on the idps itself:
Account creation allowed, when this is enabled, and the registration in general is disabled, it should work, as you wish
<img width="685" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/e802887c-1e6b-4170-88f8-c4eb7e16bdfc">

**Reply by swratten** [reply]
This seems to not give me the correct outcome unfortunately.
I disabled registration in the Organization, and Added Google IDP,  with Account creation allowed and linking allowed.

I get sent here first time logging in.

<img width="546" alt="Screen Shot 2023-10-26 at 12 26 43 PM" src="https://github.com/zitadel/zitadel/assets/4234090/993f09a4-e5fd-4116-bf69-50b0abdb1f54">

When I press register I get sent back here in infinite loop.

<img width="433" alt="Screen Shot 2023-10-26 at 12 28 26 PM" src="https://github.com/zitadel/zitadel/assets/4234090/b6baac3a-c0a6-4c30-a41f-03887a87be5d">

**Reply by swratten** [reply]
Also I have made sure External IDP allowed is enabled. And the Google IDP is "active" in the organization.
I'm running self hosted v2.34.0 through Docker

**Reply by hifabienne** [reply]
hei @swratten 
Can you tell a little bit more about your setup?
We recently pusblished a technical advisory: https://zitadel.com/docs/support/advisory/a10003 
Are you running on the older version where the instance settings get triggered? Or on the newer version where the default org settings get triggered?
Depending on that can you tell me what you have configured on the instance and organization?
And how does your authorization request look like? Do you send any special scopes?

**Comment by avdhoot** [comment]
@dedene Are you able to figured it out ? I am also interested in this use case.

**Comment by krausmaximilian** [comment]
Is there any update here?

**Reply by rajatcing** [reply]
hey @krausmaximilian No, for now,the Admin Console doesn't offer a built‑in setting to cap the IDPs a customer (organisation).
